// wot-api/wot-api.js
var errorResponse = require("./response/error.js");
var httpClient = require("request");
var clc = require('cli-color');

var WotApi = {
    config: {},

    go: function (url, form, callback) {
        console.log(clc.white("Call API: "), clc.cyan(url), form);
        form.application_id = this.config.api.application_id;
        httpClient.post(url, {form: form}, function (error, response, body) {
            if (!error && response.statusCode == 200) {
                try {
                    var parsedBody = JSON.parse(body);
                    console.log(
                        clc.white("Status: ") + clc.green(parsedBody.status),
                        clc.white("Count: ") + clc.green(parsedBody.count)
                    );
                    callback(parsedBody);
                } catch (e) {
                    console.log(clc.red("Error parsing result: "), e, body);
                    callback(errorResponse("Invalid result format."));
                }
            } else {
                console.log(clc.red("Error: "), error);
                callback(errorResponse(error.message));
            }
        });
    },
    clan: function (id, callback) {
        var url = this.config.api.host + "/clan/info/",
            form = {clan_id: id};
        this.go(url, form, callback);
    },
    account: function (id, callback) {
        var url = this.config.api.host + "/account/info/",
            form = {account_id: id};
        this.go(url, form, callback);
    },
    login: function (callback) {
        var url = this.config.api.host + "/auth/login/",
            form = {
                redirect_uri: this.config.login.resultUrl,
                nofollow: 1
            };
        this.go(url, form, callback);
    },
    login_extend: function (access_token, callback) {
        var url = this.config.api.host + "auth/prolongate/"
        this.go(url, {access_token: access_token}, callback);
    }


};
module.exports = WotApi;
