// wot-api/wot-api.js
var errorResponse = require("./response/error.js");
var httpClient = require("request");
var clc = require('cli-color');

var WotApi = {
    host: "",
    application_id: "",
    url: "",
    form: {},
    go: function(callback) {
        console.log(clc.white("Call API: "), clc.cyan(this.url), this.form);
        this.form.application_id = this.application_id;
        httpClient.post(this.url, {form: this.form}, function (error, response, body){
            if (!error && response.statusCode == 200) {
                try {
                    var parsedBody = JSON.parse(body);
                    console.log(
                            clc.white("Status: ") + clc.green(parsedBody.status),
                            clc.white("Count: ") + clc.green(parsedBody.count)
                    );
                    callback(parsedBody);
                } catch (e) {
                    console.log(clc.red("Error parsing result: "), e, body);
                    callback(errorResponse("Invalid result format."));
                }
            } else {
                console.log(clc.red("Error: "), error);
                callback(errorResponse(error.message));
            }
        });
    },
    clan: function(id, callback) {
        this.url = this.host + "/clan/info/";
        this.form = {clan_id: id};
        this.go(callback);
    },
    account: function(id, callback) {
        this.url = this.host + "/account/info/";
        this.form = {account_id: id};
        this.go(callback);
    },
    login: function(params, callback) {
        this.url = this.host + "/auth/login/";
        this.form = params;
        this.go(callback);
    }


};
module.exports = WotApi;
